Structure_Engine/Create/Loads/BarVaryingDistributedLoad/BarVaryingDistributedLoad.cs

```diff
@@ -38,36 +39,52 @@ public static partial class Create
         /**** Public Methods                            ****/
         /***************************************************/
 
-
         [Description("Creates a varying distributed load to be applied to Bar elements.")]
         [InputFromProperty("loadcase")]
         [InputFromProperty("group", "Objects")]
-        [InputFromProperty("distanceFromA")]
-        [InputFromProperty("distanceFromB")]
-        [InputFromProperty("forceA")]
-        [InputFromProperty("forceB")]
-        [InputFromProperty("momentA")]
-        [InputFromProperty("momentB")]
+        [InputFromProperty("startPosition")]
+        [InputFromProperty("endPosition")]
+        [InputFromProperty("forceAtStart")]
+        [InputFromProperty("forceAtEnd")]
+        [InputFromProperty("momentAtStart")]
+        [InputFromProperty("momentAtEnd")]
+        [InputFromProperty("relativePositions")]
         [InputFromProperty("axis")]
         [InputFromProperty("projected")]
         [Input("name", "The name of the created load.")]
         [Output("barVarLoad", "The created BarVaryingDistributedLoad.")]
-        public static BarVaryingDistributedLoad BarVaryingDistributedLoad(Loadcase loadcase, BHoMGroup<Bar> group, double distanceFromA = 0, Vector forceA = null, Vector momentA = null, double distanceFromB = 0, Vector forceB = null, Vector momentB = null, LoadAxis axis = LoadAxis.Global, bool projected = false, string name = "")
+        public static BarVaryingDistributedLoad BarVaryingDistributedLoad(Loadcase loadcase, BHoMGroup<Bar> group, double startPosition = 0, Vector forceAtStart = null, Vector momentAtStart = null, double endPosition = 1, Vector forceAtEnd = null, Vector momentAtEnd = null, bool relativePositions = true, LoadAxis axis = LoadAxis.Global, bool projected = false, string name = "")
         {
-            if ((forceA == null || forceB == null) && (momentA == null || momentB == null))
-                throw new ArgumentException("Bar varying load requires either the force at A and B and/or the moment at A and B to be defined");
+            if ((forceAtStart == null || forceAtEnd == null) && (momentAtStart == null || momentAtEnd == null))
+            {
+                Engine.Reflection.Compute.RecordError("Bar varying load requires either the force at start and end and/or the moment at start and end to be defined.");
+                return null;
+            }
```

Comment:
I think this requires a rethink, because if you have:
```
momentAtStart = {0, 0, 0};
momentAtEnd = {-1000, 0, 0};
```

This will return an error - when it is actually a valid load.

Reply:
Apologies if this is a dumb question, I am basing it purely on the subset of code I can see following @peterjamesnugent comments, and please ignore if it's derailing the conversation, but, what happens if you have a `forceAtStart` and a `forceAtEnd` but no `momentAtStart` or `momentAtEnd`? Based on the `if` statement here it appears that this would be acceptable, but the error message says it requires `either the force at start and end and/or the moment at start and end`.

Reading the error message suggests having `forceAtStart` and `forceAtEnd` but no `momentAtStart/End` is problematic, I think because of the use of `and/or` - reading it with the `and` suggests all the options are needed, but reading with the `or` suggests (what I suspect is the desired effect) that either `forceAtStart/End` OR `momentAtStart/End` are needed but not both.

Maybe the error message could be improved with commas rather than an `and` to read something like:

`Bar varying load requires either the force at start and end, or the moment at start and end, to be defined`

As I say though, very much only based on reading 3 lines highlighted in @peterjamesnugent snippet above, have not read the rest of the code, don't even understand overly what it's doing, so please do feel free to entirely ignore this late night rambling :smile:

Reply:
@peterjamesnugent , good found. Was actually no problem with the logic here, but was a problem with wrong parameters being passed from the create method accepting an `IEnumerable<Bar>` 🤦 

Have fixed that issue now, very glad that got caught!